src/processors/universal.ts

```diff
@@ -370,6 +372,162 @@ class UniversalProcessor extends BaseOrderProcessor {
+      if (giveChain === ChainId.Solana || this.chainId === ChainId.Solana) {
+        // execute unlock for each order(solana doesnt support batch unlock now)
+        for (const orderId of orderIds) {
+          const order = this.ordersDataMap.get(orderId)!;
+          const fees = await this.getFee(order, context);
+          const executionFeeAmount =
+            await context.config.client.getAmountToSend(
+              this.chainId,
+              giveChain,
+              fees.executionFees.total,
+              this.context.takeChain.fulfullProvider.connection as Web3
+            );
+          logger.debug(`executionFeeAmount = ${executionFeeAmount.toString()}`);
+          const unlockTx = await this.createOrderUnlockTx(
+            orderId,
+            order,
+            beneficiary,
+            executionFeeAmount,
+            fees,
+            context,
+            logger
+          );
+
+          const txUnlock =
+            await this.context.takeChain.unlockProvider.sendTransaction(
+              unlockTx,
+              {
+                logger,
+              }
+            );
+          logger.info(`unlock transaction ${txUnlock} is completed`);
+        }
+      } else {
+        const order = this.ordersDataMap.get(orderIds[0])!;
+        const fees = await this.getFee(order, context);
+        const executionFeeAmount = await context.config.client.getAmountToSend(
+          this.chainId,
+          giveChain,
+          fees.executionFees.total,
+          this.context.takeChain.fulfullProvider.connection as Web3
+        );
+        logger.debug(`executionFeeAmount = ${executionFeeAmount.toString()}`);
+        const batchUnlockTx = await context.config.client.sendBatchUnlock(
+          Array.from(orderIds),
+          giveChain,
+          this.chainId,
+          beneficiary,
+          executionFeeAmount,
+          {
+            web3: (this.context.takeChain.unlockProvider as EvmProviderAdapter)
+              .connection,
+            loggerInstance: createClientLogger(logger),
+            reward1: 0,
+            reward2: 0,
+          }
+        );
+
+        const txUnlock =
+          await this.context.takeChain.unlockProvider.sendTransaction(
+            batchUnlockTx,
+            {
+              logger,
+            }
+          );
+
+        logger.info(
+          `unlock for ${JSON.stringify(
+            Array.from(orderIds)
+          )} orders ${txUnlock} is completed`
+        );
+      }
+
+      // clean executed orders form queue
+      orderIds.forEach((id) => {
+        this.unlockBatchesOrderIdMap.get(giveChain)!.delete(id);
+        this.ordersDataMap.delete(id);
+      });
+
+      // check a full of batch
+      if (
+        this.unlockBatchesOrderIdMap.get(giveChain)!.size >=
+        this.params.batchUnlockSize
+      ) {
+        this.processUnlockBatches(giveChain, context);
+      } else {
+        logger.debug("Batch is not full");
```

Comment:
What is the purpose of this `if` branch? Isn't it better to use the generic code from the `else` branch where you check all possible chains?